Lib/test/test_fileinput.py

```diff
@@ -530,6 +531,22 @@ def test_iteration_buffering(self):
             self.assertRaises(StopIteration, next, fi)
             self.assertEqual(src.linesread, [])
 
+    def test_pathlib_file(self):
+        t1 = None
+        try:
+            t1 = Path(writeTmp(1, ["Pathlib file."]))
+            fi = FileInput(t1)
```

Comment:
nit: the other tests use `with FileInput(...)` so it gets closed even if an assertion fails.